tokio-util/src/sync/cancellation_token.rs

```diff
@@ -287,6 +287,22 @@ impl CancellationToken {
         }
         .await
     }
+
+    /// Runs a future to completion and returns its result wrapped inside of an `Option`
+    /// unless the `CancellationToken` is cancelled. In that case the function returns
+    /// `None` and the future gets dropped.
+    ///
+    /// The function takes self by value.
+    ///
+    /// # Cancel safety
+    ///
+    /// This method is only cancel safe if `fut` is cancel safe.
```

Comment:
The docs should mention the difference to `run_until_cancelled` in some way.
Maybe add something like this to the docs:
``` 
///
/// The function takes self by value and returns a future that owns the
/// token.
```